### ISSUE TITLE
Appearance of artefacts on the screen

### ISSUE THREAD

**Issue by TerrikG1** [QubesOS/qubes-issues#5735]
**Qubes OS version**
R4.0.3

**Affected component(s) or functionality**
Xfce, GUI

**Brief summary**
Artefacts on the screen, as well as twitching graphics. Small stripes. Slightly disappears when deployed to full screen. After a reboot, the artefacts on the screen also remain the same. When you connect the monitor to another computer, artefacts remain for a while. The problem only occurs with TemplateVM Whonix. With AppVM Whonix, as well as other TemplateVM/AppVM i have no problems.

**To Reproduce**
_Steps to reproduce the behavior:_
1) Clean installation of Qubes os
2) Running whonixcheck and console simultaneously in whonix-gw-15/whonix-ws-15. After immediately appear artefacts.

**Expected behavior**
Normal GUI behavior.

**Actual behavior**
See above.

**Screenshots**
When using PrintScreen, artefacts are not visible in the photo. So I took a picture on the phone.

![1](https://user-images.githubusercontent.com/62355676/77003498-8bbd9d00-696e-11ea-80af-0f1723f27908.jpg)
![2](https://user-images.githubusercontent.com/62355676/77003501-8ceeca00-696e-11ea-9f05-b1a6afdc0c32.jpg)
![3](https://user-images.githubusercontent.com/62355676/77003502-8d876080-696e-11ea-9380-c88ad3cf5cc8.jpg)
![4](https://user-images.githubusercontent.com/62355676/77003506-8eb88d80-696e-11ea-9bea-20b667ca286c.jpg)
![5](https://user-images.githubusercontent.com/62355676/77003508-8f512400-696e-11ea-82a4-bc2b0d61d1bb.jpg)

**Solutions you've tried**
I turn off the computer for a while. After switching on, artefacts completely disappear. If me repeat the actions, sometimes small tearing appear, but they are not as significant as artefacts during playback after a clean installation.

**Comment by TerrikG1** [comment]
When me change the screen resolution to a smaller one, the problems disappear. The original resolution is 1920x1080 - I need it.

**Comment by TerrikG1** [comment]
After some checks, I noticed these errors. Problem with the nouveau driver? Nvidia graphics card.
These errors are in QubesOS 4.0.1/4.0.2 /4.0.3. In versions 4.0 and 3.2.1 there are no such problems.

**glxinfo | grep OpenGL**
```
nvc0_screen_create:953 - Error allocating PGRAPH context for 3D: -22
libGL error: failed to create dri screen
libGL error: failed to load driver: nouveau
```

**grep -e EE -e WW /var/log/Xorg.0.log**
```
(WW) warning, (EE) error, (NI) not implemented, (??) unknown.
[    27.534] (WW) Warning, couldn't open module nv
[    27.534] (EE) Failed to load module "nv" (module does not exist, 0)
[    27.537] (WW) Falling back to old probe method for modesetting
[    27.537] (WW) Falling back to old probe method for fbdev
[    27.537] (WW) Falling back to old probe method for vesa
[    28.037] (EE) AIGLX error: Calling driver entry point failed
[    28.038] (EE) AIGLX: reverting to software rendering
```
**LIBGL_DEBUG=verbose glxinfo | grep -i "opengl version"**
```
libGL: OpenDriver: trying /usr/lib64/dri/tls/nouveau_dri.so
libGL: OpenDriver: trying /usr/lib64/dri/nouveau_dri.so
nvc0_screen_create:953 - Error allocating PGRAPH context for 3D: -22
libGL error: failed to create dri screen
libGL error: failed to load driver: nouveau
libGL: OpenDriver: trying /usr/lib64/dri/tls/swrast_dri.so
libGL: OpenDriver: trying /usr/lib64/dri/swrast_dri.so
libGL: Can't open configuration file /home/user/.drirc: No such file or directory.
libGL: Can't open configuration file /home/user/.drirc: No such file or directory.
OpenGL version string: 2.1 Mesa 17.0.5
```
**grep -i aiglx /var/log/Xorg.0.log**
```
[    28.037] (EE) AIGLX error: Calling driver entry point failed
[    28.038] (EE) AIGLX: reverting to software rendering

```

**Comment by TerrikG1** [comment]
The Qubes version does not affect the appearance of artifacts. I have experimented with various versions of QubesOS since version 3.2.1 and settled on 4.0.1.

My actions were as follows:
1) Install Qubes 4.0.1
2) Updated dom0 to the latest version 4.0.3
3) Checked Whonix TemplateVm (version 14) - no artifacts found.
4) Updated Whonix TemplateVM to version 15
5) I checked Whonix TemplateVM (version 15) and was able to reproduce previously discovered artifacts.

The problem with TemplateVM (whonix), namely with the latest version (15). With other TemplateVM (Debian, Fedora) artifacts were not noticed. I wrote earlier that appvm whonix did not inherit the same artifacts. But after my experiments, I was able to reproduce artifacts on AppVM Whonix.

And the question about the errors above, namely:

1) nvc0_screen_create: 953 - Error allocating PGRAPH context for 3D: -22
2) libGL error: failed to create dri screen
3) libGL error: failed to load driver: nouveau

Are they significant? Can I fix them somehow?
